altos-rust/port/cortex-m0/src/interrupt/defs.rs

```diff
@@ -0,0 +1,81 @@
+// interrupt/defs.rs
+// AltOSRust
+//
+// Created by Mohammed Inoue on 2/21/17
+
+
+#[derive(Copy, Clone)]
+pub enum Hardware {
+    WWDG,
+    PVDVDDIO2,
+    RTC,
+    FLASH,
+    RCCCRS,
+    EXTI01,
+    EXTI23,
+    EXTI415,
+    TSC,
+    DMACH1,
+    DMACH23,
+    DMACH4PLUS,
+    ADCCOMP,
+    TIM1BRKUP,
+    TIM1CC,
+    TIM2,
+    TIM3,
+    TIM6,
+    TIM7,
+    TIM14,
+    TIM15,
+    TIM16,
+    TIM17,
+    I2C1,
+    I2C2,
+    SPI1,
+    SPI2,
+    USART1,
+    USART2,
+    USART3PLUS,
+    CECCAN,
+    USB,
+}
+
+pub fn map_hardware(target:Hardware) -> u8 {
+    match target {
```

Comment:
Instead of using this function, you can actually just map each enum to an integer representation directly.
```rust
WWDG = 0
PVDVDIO2 = 1
RTC = 2
// etc...
```

Then when you need the integer representation you can cast it at the usage site.